src/GafferSceneUI/SceneView.cpp

```diff
@@ -1335,77 +1335,21 @@ void SceneView::frame( const GafferScene::PathMatcher &filter, const Imath::V3f
 
 void SceneView::expandSelection( size_t depth )
 {
-	Context::Scope scopedContext( getContext() );
-
 	PathMatcher &selection = const_cast<GafferScene::PathMatcherData *>( m_sceneGadget->getSelection() )->writable();
-	PathMatcher &expanded = expandedPaths()->writable();
 
-	std::vector<string> toExpand;
-	selection.paths( toExpand );
+	PathMatcher newlyExpandedLeaves = ContextAlgo::expandDescendants( getContext(), selection, preprocessedInPlug<ScenePlug>(), depth - 1 );
```

Comment:
I definitely am not positive on this one, but that was intentional in an effort to get both `ContextAlgoTest` and `SceneViewTest` passing. I somehow justified it to myself on the basis that the semantics of `ContextAlgo::expandDescendants` and `SceneView::expandSelection` are starting from different points in the intended ancestry.

That might all be a bit of a mind-bender on my part though. Would you mind having a close look at `ContextAlgoTest.testExpandedPaths` to see if you think the logic I'm testing for is correct? I got a bit twisted up and re-wrote those tests a handful of times. I thought I'd ended in a situation that makes sense, but now I'm not so sure.